org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalReplacementProvider.java

```diff
@@ -937,13 +939,15 @@ private ITypeBinding getExpectedTypeForGenericParameters() {
 		for (int i= 0; i < keys.length; i++) {
 			keys[i]= String.valueOf(chKeys[0]);
 		}
-
+		// https://github.com/eclipse/eclipse.jdt.ls/pull/2535
+		if (bindings.get(keys[0]) instanceof ITypeBinding) {
+			return (ITypeBinding) bindings.get(keys[0]);
+		}
```

Comment:
To keep consistent as previous implementation, it can
```java
		if (bindings.size() > 0) {
			return (ITypeBinding) bindings.get(keys[0]);
		}
```

Reply:
Fixed.